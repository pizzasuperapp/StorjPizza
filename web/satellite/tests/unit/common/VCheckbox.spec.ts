// Copyright (C) 2019 Storj Labs, Inc.
// See LICENSE for copying information.

import Checkbox from '@/components/common/VCheckbox.vue';

import { mount, shallowMount } from '@vue/test-utils';

describe('Checkbox.vue', () => {
    it('renders correctly', () => {
        const wrapper = shallowMount(Checkbox);

        expect(wrapper).toMatchSnapshot();
    });

    it('emit setData on change correctly', () => {
        const wrapper = mount(Checkbox);

        wrapper.find('input').trigger('change');
        wrapper.find('input').trigger('change');

        const emittedSetData = wrapper.emitted('setData');
        if (emittedSetData) expect(emittedSetData.length).toEqual(2);
    });

    it('emits with data correctly', () => {
        const wrapper = mount(Checkbox);

        wrapper.vm.$emit('setData', true);

        const emittedSetData = wrapper.emitted('setData');
        if (emittedSetData) expect(emittedSetData[0][0]).toEqual(true);
    });

    it('renders correctly with error', () => {
        const wrapper = shallowMount(Checkbox, {
            propsData: {isCheckboxError: true},
        });

        expect(wrapper).toMatchSnapshot();
    });
});
