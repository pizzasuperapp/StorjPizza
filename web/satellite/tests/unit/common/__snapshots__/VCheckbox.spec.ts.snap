// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Checkbox.vue renders correctly 1`] = `
<div class="wrap"><label class="container"><input id="checkbox" type="checkbox" class="checkmark-input"> <span class="checkmark"></span></label>
  <!---->
</div>
`;

exports[`Checkbox.vue renders correctly with error 1`] = `
<div class="wrap"><label class="container"><input id="checkbox" type="checkbox" class="checkmark-input"> <span class="checkmark error"></span></label>
  <!---->
</div>
`;
